colcon_sanitizer_reports/sanitizer_log_parser.py

```diff
@@ -104,6 +110,38 @@ def get_csv(self) -> str:
 
         return csv_f_out.getvalue()
 
+    def get_xml(self) -> str:
```

Comment:
This pattern:
```py
def f():
  # comment
  some code

  # comment
  some code

  # comment
  some code
```

...indicates that `f()` could be divided into multiple sub-functions.

Please consider splitting this function into multiple smaller functions or methods which will be easier to test.

Reply:
Good point. Split this function into its own class and separate unit tests.